### PR TITLE
Experimental ARMv7-R support

### PR DESCRIPTION
**Pull request by Hoverbear**
Adds `armv7r-none-eabihf` and `armebv7r-none-eabihf` as experimental targets. Note experimental targets are not qualified and untested.

### CODE REVIEW

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try-

**Comment by Hoverbear** [comment]
bors try

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -379,7 +379,7 @@ jobs:
     resource_class: macos.m1.medium.gen1 # 4 cores
     environment:
       FERROCENE_HOST: aarch64-apple-darwin
-      FERROCENE_TARGETS: aarch64-apple-darwin,x86_64-apple-darwin,aarch64-unknown-none,thumbv7em-none-eabi,thumbv7em-none-eabihf,armv8r-none-eabihf,wasm32-unknown-unknown
+      FERROCENE_TARGETS: aarch64-apple-darwin,x86_64-apple-darwin,aarch64-unknown-none,thumbv7em-none-eabi,thumbv7em-none-eabihf,armv8r-none-eabihf,wasm32-unknown-unknown,,armv7r-none-eabihf,armebv7r-none-eabihf
```
Comment (Hoverbear) [inline .circleci/workflows.yml:382]:
```suggestion
      FERROCENE_TARGETS: aarch64-apple-darwin,x86_64-apple-darwin,aarch64-unknown-none,thumbv7em-none-eabi,thumbv7em-none-eabihf,armv8r-none-eabihf,wasm32-unknown-unknown,armv7r-none-eabihf,armebv7r-none-eabihf
```

File: ferrocene/doc/release-notes/src/next.rst [hunk 2]
```diff
@@ -20,6 +20,8 @@ shipped as a preview.
   new target is:
```
Comment (pietroalbini) [inline ferrocene/doc/release-notes/src/next.rst:20]: Plural :)

**Review by pietroalbini — APPROVED** [review]
bors merge

**Comment by Hoverbear** [comment]
Rebased, also added a missing entry into `next.rst`

**Comment by Hoverbear** [comment]
(Rebased)

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**